newIDE/app/src/ObjectGroupsList/index.js

```diff
@@ -243,22 +244,30 @@ export default class GroupsListContainer extends React.Component<Props, State> {
     }
   };
 
-  _setAsGlobalGroup = (groupWithContext: GroupWithContext) => {
+  _setAsGlobalGroup = (i18n: I18nType, groupWithContext: GroupWithContext) => {
     const { group } = groupWithContext;
-    const { globalObjectGroups, objectGroups } = this.props;
+    const { globalObjectGroups, objectGroups, onSetAsGlobalGroup } = this.props;
 
     const groupName = group.getName();
 
     if (globalObjectGroups.has(groupName)) {
       showWarningBox(
-        'A global object with this name already exists. Please change the object name before setting it as a global object',
+        i18n._(
+          t`A global object with this name already exists. Please change the object name before setting it as a global object`
+        ),
         { delayToNextTick: true }
       );
       return;
     }
 
+    if (onSetAsGlobalGroup && !onSetAsGlobalGroup(groupName)) {
```

Comment:
It took me a while to understand the logic.
Usually we use functions `onXXX` to trigger them **after** the whole logic is done, whereas here, you use it before.
Maybe a different name? `beforeSetAsGlobalGroup`?

Reply:
Yes, good idea!